### ISSUE TITLE
Training blocks rendering/halts CSS

### ISSUE THREAD

**Issue by skulldoggery** [skulldoggery/gamutt#12]
Training should happens asynchronously from re-rendering the DOM, if possible. Promises and async effects should let us do this, but I'm still getting my hands around some fundamentals with SolidJS. 

Blocking is negligible unless the neural network fails to reach its target error threshold. If async training isn't an option, we'll need to make one or more of the following changes, which will negatively impact Gamutt's predictions: 

- Lower iterations
  - Linear improvement to runtime if error threshold can't be reached
- Raise error threshold
  - More likely to have out of range predictions, but we're already clamping the output between 0 and 1
- Increase learning rate
  - May cause over fitting, but not necessarily badb
